### PR TITLE
Various console redesign fixes

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR applies various fixes to the console redesign.

#### Changes

- Fix browser console warnings and errors
- Fix top entities panel responsiveness

Screenshot 2024-05-24 at 16 28 24

Screenshot 2024-05-24 at 16 28 36

##### Steps

- Run the console
- Resize the window to see the top entities panel responsive
- open browser console and see no errors

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/store/selectors/applications.js [hunk 1]
```diff
@@ -66,14 +68,19 @@ export const selectApplicationDevEUICount = state =>
...
+  (ids, apps) => ids.map(id => apps[id]),
+)
 export const selectApplicationsTotalCount = state => selectAppsTotalCount(state)
-export const selectApplicationsWithDeviceCounts = state =>
-  selectApplications(state).map(app => ({
-    ...app,
-    _devices: selectApplicationDeviceCount(state, app.ids.application_id),
-  }))
+export const selectApplicationsWithDeviceCounts = createSelector(
+  [selectApplications, selectApplicationStore],
```
Comment (kschiffer) [inline pkg/webui/console/store/selectors/applications.js]:
```suggestion
  [selectApplications, selectApplicationDeviceCount],
```

File: pkg/webui/console/containers/top-entities-dashboard-panel/item.js [hunk 2]
```diff
@@ -71,7 +71,14 @@ const EntitiesItem = ({ bookmark, headers, last }) => {
               [styles.entityCellSmall]: headers[index].name === 'type',
             })}
           >
-            {headers[index].render(value, entityID)}
+            <div
+              className={classNames({
+                [styles.entityCellContent]: headers[index].name === 'name',
+                [styles.entityCellLastColumn]: headers[index].name === 'lastSeen',
```
Comment (kschiffer) [inline pkg/webui/console/containers/top-entities-dashboard-panel/item.js]: Why not `:last-child`?

File: pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl [hunk 3]
```diff
@@ -44,4 +44,15 @@
...
+
+@container panel (max-width: 360px)
+  .entity-cell-content
+    width: 35%
+    overflow: hidden
+    text-overflow: ellipsis
+  .entity-cell-last-column
+    position: absolute
+    top: 20px
+    right: 10px
```
Comment (kschiffer) [inline pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: Please remove this and refactor the panel to not use infinite scroll but a regular scroll with a limit of 10 entities.

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/components/table/cell/cell.styl [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 .cell
   display: table-cell
-  padding: 0 $cs.s
+  padding: 0 0 0 $cs.xs
```
Comment (kschiffer) [inline pkg/webui/components/table/cell/cell.styl]: That's not a good solution. There should still be padding to both sides, just a bit less.

```css
.cell
  …
  padding: 0 calc($cs.s / 2)
  &:first-child
    padding: 0 calc($cs.s / 2) 0 $cs.s
  &:last-child
    padding: 0 $cs.s 0 calc($cs.s / 2)

File: pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl [hunk 5]
```diff
@@ -17,31 +17,30 @@
...
-      width: 2.5rem
\ No newline at end of file
+
+    &-header
+      top: 0
+      position: sticky
+      background: var(--c-bg-neutral-min)
+      z-index: 1
+
+.scroll-gradient
```
Comment (kschiffer) [inline pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: You should be able to use the new `<ScrollFader />` component to do this.

File: pkg/webui/lib/components/require-request.js [hunk 6]
```diff
@@ -54,7 +54,7 @@ const RequireRequest = React.memo(
 )
 
 RequireRequest.propTypes = {
-  children: PropTypes.node.isRequired,
+  children: PropTypes.node,
```
Comment (kschiffer) [inline pkg/webui/lib/components/require-request.js]: What was the reason to add remove this? Seems a bit odd.

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/lib/components/require-request.js (hunk 6 above)
Comment (ryaplots) [reply pkg/webui/lib/components/require-request.js]: In the status badge component there is the following structure:
```
<RequireRequest requestAction={getNetworkStatusSummary()} handleErrors={false}>
      {status && (
        ...
      )}
</RequireRequest>
```

If the status is undefined, the require request will have no children and therefore there will be a type error in the browser.

File: pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl (hunk 5 above)
Comment (ryaplots) [reply pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: I am having a hard time using <ScrollFader /> for several reasons:
1. The heights are too small and the scroll is not visible, so I tried to increase those but it breaks the layout.
2. The opacity simply does not update on scroll, if I put it outside the table:
```
<ScrollFader className={styles.scrollGradient} ref={listRef} light>
    <Table>
        <Table.Head className={styles.topEntitiesPanelOuterTableHeader}>{columns}</Table.Head>
        <Table.Body emptyMessage={m.empty}>{rows}</Table.Body>
     </Table>
</ScrollFader>
        
```
3. Putting the fader inside the body breaks the table:
```
    <Table>
        <Table.Head className={styles.topEntitiesPanelOuterTableHeader}>{columns}</Table.Head>
        <ScrollFader className={styles.scrollGradient} ref={listRef} light>
              <Table.Body emptyMessage={m.empty}>{rows}</Table.Body>
        </ScrollFader>
     </Table>
```

What is the `fadeHeight` meant for?
Am I missing something here?

Comment (kschiffer) [reply pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: First implementation is the right one. Guess I'd need to put an eye on this myself. Can you push the commit?

File: pkg/webui/lib/components/require-request.js (hunk 6 above)
Comment (kschiffer) [reply pkg/webui/lib/components/require-request.js]: `{ status ? (…) : null }`  
Should work I believe.

File: pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl (hunk 5 above)
Comment (ryaplots) [reply pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: done

File: pkg/webui/lib/components/require-request.js (hunk 6 above)
Comment (ryaplots) [reply pkg/webui/lib/components/require-request.js]: This results solution + adding back the `isRequired` results in `Warning: Failed prop type: The prop 'children' is marked as required in '<<anonymous>>', but its value is 'null'.` I will try to find a better solution

File: pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl (hunk 5 above)
Comment (ryaplots) [reply pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: I looked into this again and I found a solution. Not sure it's the best one tho

Comment (kschiffer) [reply pkg/webui/console/containers/top-entities-dashboard-panel/top-entities-panel.styl]: I've applied a fix myself, should be good now.

**Review by kschiffer — APPROVED** [review]

**Pull request merged**